src/lineup/ALineUpView2.ts

```diff
@@ -221,6 +231,112 @@ export abstract class ALineUpView2 extends AView {
     this.handleSelectionColumnsImpl(selection);
   }
 
+  private getColumnColor(id: number) : string {
+    let color = '';
+    if(!this.colorMap.has(id)) {
+      color = this.colors.shift();
+      this.colorMap.set(id, color);
+    } else {
+      color = this.colorMap.get(id);
+    }
+    return color;
+  }
+
+  private freeColumnColor(id: number) : void {
+    const color = this.colorMap.get(id);
+    if(color) {
+      this.colorMap.delete(id);
+      if(this.colors.indexOf(color) === -1) {
+        this.colors.unshift(color);
+      }
+    }
+  }
+
+  private addDynamicColumns(ids: number[]) : void {
+    const ranking = this.lineup.data.getLastRanking();
+    const usedCols = ranking.flatColumns.filter((col) => (<any>col.desc).selectedSubtype !== undefined);
+    const dynamicColumnIDs = new Set<string>(usedCols.map((col) => `${(<any>col.desc).selectedId}_${(<any>col.desc).selectedSubtype}`));
+
+    const addColumn = (desc: IAdditionalColumnDesc, newColumnPromise: Promise<IScoreRow<any>[]>, id: number) => {
+      //mark as lazy loaded
+      (<any>desc).lazyLoaded = true;
+      this.withoutTracking(() => {
+        this.addColumn(desc, newColumnPromise, id, true); // true == withoutTracking
+      });
+    };
+
+    ids.forEach((id) => {
+      this.getSelectionColumnDesc(id)
+        .then((columnDesc) => {
+          // add multiple columns
+          if(Array.isArray(columnDesc)) {
+            if(columnDesc.length > 0) {
+              // Save which columns have been added for a specific element in the selection
+              const selectedElements = new Set<string>(columnDesc.map((desc) => `${id}_${desc.selectedSubtype}`));
+
+              // Check which items are new and should therefore be added as columns
+              const addedParameters = set_diff(selectedElements, dynamicColumnIDs);
+
+              if(addedParameters.size > 0) {
+                // Filter the descriptions to only leave the new columns and load them
+                const columnsToBeAdded = columnDesc.filter((desc) => addedParameters.has(`${id}_${desc.selectedSubtype}`));
+                const newColumns: any = this.loadSelectionColumnData(id, columnsToBeAdded);
+
+                // add new columns
+                newColumns.then((dataPromise) => {
+                  columnsToBeAdded.forEach((desc, i) => {
```

Comment:
would be more efficient to do this `withoutTracking()` outside the loop, since it is a costly operation